dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+     */
+    @Override
+    public Mono<UpdateResponse> update(HttpHeaders headers, UpdateRequest updateRequest) {
+        return sendRequest(updateRequest, DotReactiveElasticsearchClient.RequestCreator.update(), UpdateResponse.class, headers).publishNext();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.http.HttpHeaders, org.elasticsearch.action.delete.DeleteRequest)
+     */
+    @Override
+    public Mono<DeleteResponse> delete(HttpHeaders headers, DeleteRequest deleteRequest) {
+
+        return sendRequest(deleteRequest, DotReactiveElasticsearchClient.RequestCreator.delete(), DeleteResponse.class, headers) //
+                .publishNext();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#count(org.springframework.http.HttpHeaders, org.elasticsearch.action.search.SearchRequest)
+     */
+    @Override
+    public Mono<Long> count(HttpHeaders headers, CountRequest countRequest) {
+        return sendRequest(countRequest, DotReactiveElasticsearchClient.RequestCreator.count(), CountResponse.class, headers) //
+                .map(CountResponse::getCount) //
+                .next();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.http.HttpHeaders, org.elasticsearch.action.search.SearchRequest)
+     */
+    @Override
+    public Flux<SearchHit> search(HttpHeaders headers, SearchRequest searchRequest) {
+
+        return sendRequest(searchRequest, DotReactiveElasticsearchClient.RequestCreator.search(), SearchResponse.class, headers) //
+                .map(SearchResponse::getHits) //
+                .flatMap(Flux::fromIterable);
+    }
+
+    public SearchResponse search(SearchRequest searchRequest, RequestOptions options){
+        return sendRequest(searchRequest, DotReactiveElasticsearchClient.RequestCreator.search(), SearchResponse.class, HttpHeaders.EMPTY).blockFirst();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#scroll(org.springframework.http.HttpHeaders, org.elasticsearch.action.search.SearchRequest)
+     */
+    @Override
+    public Flux<SearchHit> scroll(HttpHeaders headers, SearchRequest searchRequest) {
+
+        TimeValue scrollTimeout = searchRequest.scroll() != null ? searchRequest.scroll().keepAlive()
+                : TimeValue.timeValueMinutes(1);
+
+        if (searchRequest.scroll() == null) {
+            searchRequest.scroll(scrollTimeout);
+        }
+
+        EmitterProcessor<ActionRequest> outbound = EmitterProcessor.create(false);
+        FluxSink<ActionRequest> request = outbound.sink();
+
+        EmitterProcessor<SearchResponse> inbound = EmitterProcessor.create(false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'inbound' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)